File: app/src/main/java/com/example/pam_project/landing/SharedPreferencesFtuStorage.java
package com.example.pam_project.landing;

import android.content.SharedPreferences;

public class SharedPreferencesFtuStorage implements FtuStorage {

    private static final String FTU_KEY = "is_ftu";
    private final SharedPreferences sharedPref;

    public SharedPreferencesFtuStorage(final SharedPreferences sharedPref) {
        this.sharedPref = sharedPref;
    }

    @Override
    public boolean isActive() {
        return sharedPref.getBoolean(FTU_KEY, true);
    }

    @Override
    public void deactivate() {
        sharedPref.edit().putBoolean(FTU_KEY, false).apply();
    }
}

File: app/src/main/java/com/example/pam_project/landing/SharedPreferencesFtuStorage.kt
package com.example.pam_project.landing

import android.content.SharedPreferences

class SharedPreferencesFtuStorage(private val sharedPref: SharedPreferences) : FtuStorage {
    override val isActive: Boolean
        get() = sharedPref.getBoolean(FTU_KEY, true)

    override fun deactivate() {
        sharedPref.edit().putBoolean(FTU_KEY, false).apply()
    }

    companion object {
        private const val FTU_KEY = "is_ftu"
    }
}